test/test_roscpp/test/CMakeLists.txt

```diff
@@ -1,3 +1,5 @@
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
```

Comment:
No, this was only needed to make the tests compile... without that tests unrelated to this change didn't compile on my machine. If it is not needed with more recent ROS versions than indigo anymore, you can also remove it again.